client/my-sites/backup/backup-storage-space/backup-storage-space-upsell/index.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import React, { FunctionComponent, useCallback, useEffect } from 'react';
+import { Button } from '@wordpress/components';
+import { sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { recordTracksEvent } from 'calypso/state/analytics/actions/record';
+import { useTranslate, translate as translateType } from 'i18n-calypso';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+export type BackupStorageSpaceUpsellOptions =
+	| 'no_upsell'
+	| 'first_upsell'
+	| 'second_upsell'
+	| 'out_of_storage';
+
+const getStatusText = (
+	upsellOption: BackupStorageSpaceUpsellOptions,
+	storageLimit: number,
+	translate: typeof translateType
```

Comment:
I've never done it before, but I think TypeScript exposes a utility type called `ReturnType` for exactly this case. If it works, we could wipe out the import of `translate` as well.

```suggestion
	translate: ReturnType< typeof useTranslate >
```

Reply:
I didn't know that was a thing, but it seems to work. Good call!